workshops/fabric-e2e-rag/workshop.md

```diff
@@ -0,0 +1,480 @@
+
+```python
+from synapse.ml.services import AnalyzeDocument
+from pyspark.sql.functions import col
+
+analyze_document = (
+    AnalyzeDocument()
+    .setPrebuiltModelId("prebuilt-layout")
+    .setSubscriptionKey(ai_services_key)
+    .setLocation(ai_services_location)
+    .setImageBytesCol("content")
+    .setOutputCol("result")
+)
+
+analyzed_df = (
+    analyze_document.transform(df)
+    .withColumn("output_content", col("result.analyzeResult.content"))
+    .withColumn("paragraphs", col("result.analyzeResult.paragraphs"))
+).cache()
+```
+
+We can observe the analyzed Spark DataFrame named ```analyzed_df``` using the following code. Note that we drop the `content` column as it is not needed anymore.
+
+```python
+analyzed_df = analyzed_df.drop("content")
+display(analyzed_df)
+```
+
+---
+
+# Generating Embeddings and Storing them in a Vector Store
+
+Now that we have the text content of the PDF documents, we can generate embeddings for the text using Azure OpenAI. Embeddings are vector representations of the text that can be used to compare the similarity between different pieces of text.
+
+Before we can generate the embeddings, we need to split the text into chunks. To do this we leverage SynapseML’s PageSplitter to divide the documents into smaller sections, which are subsequently stored in the `chunks` column. This allows for more granular representation and processing of the document content.
+
+```python
+from synapse.ml.featurize.text import PageSplitter
+
+ps = (
+    PageSplitter()
+    .setInputCol("output_content")
+    .setMaximumPageLength(4000)
+    .setMinimumPageLength(3000)
+    .setOutputCol("chunks")
+)
+
+splitted_df = ps.transform(analyzed_df)
+display(splitted_df)
+```
+
+Note that the chunks for each document are presented in a single row inside an array. In order to embed all the chunks in the following cells, we need to have each chunk in a separate row. To accomplish that, we first explode these arrays so there is only one chunk in each row, then filter the Spark DataFrame in order to only keep the path to the document and the chunk in a single row.
+
+```python
+# Each column contains many chunks for the saame document as a vector.
+# Explode will distribute and replicate the content of a vecor across multple rows
+from pyspark.sql.functions import explode, col
+
+exploded_df = splitted_df.select("path", explode(col("chunks")).alias("chunk")).select(
+    "path", "chunk"
+)
+display(exploded_df)
+```
+Next we'll generate the embeddings for each chunk. To do this we utilize both SynapseML and Azure OpenAI Service. By integrating the Azure OpenAI service with SynapseML, we can leverage the power of the Apache Spark distributed computing framework to process numerous prompts using the OpenAI service. 
+
+```python
+from synapse.ml.services import OpenAIEmbedding
+
+embedding = (
```

Comment:
Here we can use the Fabric Azure AI Services deployment instead of the Azure OpenAI one